enhancements/network/ovn-k-localnet-api.md

```diff
@@ -0,0 +1,516 @@
+---
+title: user-defined-network-segmentation
+authors:
+  - "@ormergi"
+reviewers:
+  - "@maiqueb"
+  - TBD
+approvers:
+  - TBD
```

Comment:
@JoelSpeed could you review this API ? We've tried to keep it really small and focused on the API itself.

Or find someone on your team that could lend a hand.